From 985ebcf8d239324787469c082bec631c89767a2e Mon Sep 17 00:00:00 2001
From: Bian Jin chen <kenjc.bian@rock-chips.com>
Date: Sat, 12 Oct 2019 17:12:24 +0800
Subject: [PATCH] arm64: rockchip_defconfig: enable CONFIG_HW_RANDOM

Enable the Rockchip HW random driver to prevent Android init
from hang due to low entropy.

Change-Id: I0f2c2c1c27c4c91002a42c8cb77e44809815bd03
Signed-off-by: Bian Jin chen <kenjc.bian@rock-chips.com>
---
 arch/arm64/configs/rockchip_defconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index cae8e90fc62f..e5120ba5428d 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -476,7 +476,8 @@ CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_8250_NR_UARTS=6
 CONFIG_SERIAL_8250_RUNTIME_UARTS=6
 CONFIG_SERIAL_8250_DW=y
-# CONFIG_HW_RANDOM is not set
+CONFIG_HW_RANDOM=y
+CONFIG_HW_RANDOM_ROCKCHIP=y
 # CONFIG_DEVPORT is not set
 CONFIG_I2C_CHARDEV=y
 CONFIG_I2C_GPIO=y
-- 
2.35.3

